brut.apktool/apktool-lib/src/main/java/brut/androlib/res/decoder/ARSCDecoder.java

```diff
@@ -232,17 +232,17 @@ private ResBagValue readComplexEntry() throws IOException,
 
         ResValueFactory factory = mPkg.getValueFactory();
         Duo<Integer, ResScalarValue>[] items = new Duo[count];
-        ResValue resValue;
+        ResIntBasedValue resValue;
         int resId;
 
         for (int i = 0; i < count; i++) {
             resId = mIn.readInt();
             resValue = readValue();
 
-            try {
```

Comment:
The instanceof/attempt should be equivalent to the try {} catch (ClassCastException) that was there before (it passes my local tests at least). I can revert it to the previous form to simplify the review if you like, but I figured that an internal exception like that would have some performance implications.